### ISSUE TITLE
Passwords with dollar signs are improperly handled.

### ISSUE THREAD

**Issue by dustymabe** [projectatomic/adb-vagrant-registration#31]
This plugin tries to run the subscription-manager register via the shell and it passes the username/password on the command line as arguments. If your password has a $ in it then this doesn't work as the shell tries to interpret the $ as a variable. 

This may be broken for other characters other than $.

**Comment by voxik** [comment]
IMO, you should use shell escapes in such case. If we escape the password for you, then it would not be possible to use env variables if somebody wants to use them for some reason (nor sure how realistic this is though).

**Comment by dustymabe** [comment]
It would be odd for the user to have to shell escape his password. He would have to know the implementation details of the program to know how the password is eventually going to get used; not really something we should expect of the users. 

As for environment variables, I think you are referring to if the user himself somehow injected an environment var into the vagrant vm during provisioning and wanted to re-use that var? I don't think that is realistic (does it even work?).

In my case I am setting the environment variables and then using them in the Vagrantfile with config.registration.username = ENV['SUB_USERNAME']

**Comment by strzibny** [comment]
@dustymabe changed in https://github.com/projectatomic/adb-vagrant-registration/commit/07d0b66273bf76fa53143068109bc19967bee6db. Could you please try the new version (0.0.14) and tell me if this works for you?

**Comment by dustymabe** [comment]
```
[dmabe@hattop vagrant (master=)]$ vagrant plugin update 
Updating installed plugins...
Updated 'vagrant-libvirt' to version '0.0.30'!
Updated 'vagrant-registration' to version '0.0.14'!
[dmabe@hattop vagrant (master=)]$ 
[dmabe@hattop vagrant (master=)]$ 
[dmabe@hattop vagrant (master=)]$ vagrant status
/home/dmabe/.vagrant.d/gems/gems/vagrant-registration-0.0.14/lib/vagrant-registration/config.rb:13:in `finalize!': undefined method `skip' for #<Object:0x0000000146c3b8> (NoMethodError)
        from /usr/share/vagrant/lib/vagrant/config/v2/root.rb:50:in `block in finalize!'
        from /usr/share/vagrant/lib/vagrant/config/v2/root.rb:49:in `each'
        from /usr/share/vagrant/lib/vagrant/config/v2/root.rb:49:in `finalize!'
        from /usr/share/vagrant/lib/vagrant/config/v2/loader.rb:21:in `finalize'
        from /usr/share/vagrant/lib/vagrant/config/loader.rb:158:in `load'
        from /usr/share/vagrant/lib/vagrant/vagrantfile.rb:28:in `initialize'
        from /usr/share/vagrant/lib/vagrant/environment.rb:691:in `new'
        from /usr/share/vagrant/lib/vagrant/environment.rb:691:in `vagrantfile'
        from /usr/share/vagrant/lib/vagrant/environment.rb:441:in `host'
        from /usr/share/vagrant/lib/vagrant/environment.rb:207:in `block in action_runner'
        from /usr/share/vagrant/lib/vagrant/action/runner.rb:33:in `call'
        from /usr/share/vagrant/lib/vagrant/action/runner.rb:33:in `run'
        from /usr/share/vagrant/lib/vagrant/environment.rb:428:in `hook'
        from /usr/share/vagrant/lib/vagrant/environment.rb:673:in `unload'
        from /usr/share/vagrant/bin/vagrant:177:in `ensure in <main>'
        from /usr/share/vagrant/bin/vagrant:177:in `<main>'
```

Related to 7f81f9f ?

**Comment by strzibny** [comment]
@dustymabe Can you tell me more about your environment? Your Vagrantfile, host/guest systems etc?

**Comment by dustymabe** [comment]
@strzibny Vagrantfile is at [1]. My environment is Fedora 21 a bit of the setup instructions that I used are at [2]. Does that answer your question?


[1] - https://github.com/whitel/summit-2015-lab-15812/blob/master/vagrant/Vagrantfile
[2] - https://github.com/whitel/summit-2015-lab-15812/blob/master/vagrant/README.md

**Comment by strzibny** [comment]
@dustymabe Strange, can't reproduce ATM. I downloaded the git repo and running it with 0.0.14. I also have Fedora 21. Can you post all components versions? `ruby -v`, `rpm -q vagrant`, `vagrant plugin list`.

**Comment by dustymabe** [comment]
@strzibny I powered off that computer last night and am at work right now. Will get all of this information for you and post back here tonight. Thanks for looking into this.

**Comment by dustymabe** [comment]
```
[dmabe@hattop ~]$ ruby -v 
ruby 2.1.6p336 (2015-04-13 revision 50298) [x86_64-linux]
[dmabe@hattop ~]$ 
[dmabe@hattop ~]$ rpm -q vagrant
vagrant-1.7.2-5.fc21.1.noarch
[dmabe@hattop ~]$ 
[dmabe@hattop ~]$ vagrant plugin list
vagrant-libvirt (0.0.30, system)
vagrant-registration (0.0.14)
vagrant-reload (0.0.1)
[dmabe@hattop ~]$
```

One thing that might be of interest is that when I updated vagrant-registration I ran `vagrant plugin update` and it also updated vagrant-libvirt to 0.0.30 from 0.0.29.

**Comment by strzibny** [comment]
Ok, I did a completly clean environment and can reproduce. So you are right and this needs to be fixed. Thanks a lot for spotting the error!

**Comment by strzibny** [comment]
@dustymabe 0.0.15 released and 0.0.14 yanked. Now you can hopefully test this for this very issue.

**Comment by dustymabe** [comment]
Works now. Thanks
